drop-transfer/src/ws/server/mod.rs

```diff
@@ -128,8 +135,14 @@ pub(crate) fn start(
                     .map_err(|_| warp::reject::not_found())
             }))
             .and(
-                warp::filters::addr::remote().then(|peer: Option<SocketAddr>| async move {
-                    peer.expect("Transport should use IP addresses")
+                warp::filters::addr::remote().then(move |peer: Option<SocketAddr>| {
+                    let rl = rate_limiter.clone();
+                    async move {
+                        let peer = peer.expect("Transport should use IP addresses");
+
+                        rl.until_key_ready(&peer.ip()).await;
```

Comment:
Several idas in one comment:
- This looks pretty clean and neat. I imagined we will just drop the connections, as we do not drop, maybe there's a way to check the `governor` that the slot is not ready and a debug log to it so the delays could be tracked down in the logs?

- On the other hand, what do you think about dropping the retries if they reach the limit? In this leaky-bucket case it will potentially leak if we have some erroneous retry loop

- The default size of 5 is way too small IMHO, it would be at least 10x. My main concern is catching erroneous behaviour of our own code where we would spam the peer with infinite retries or something like that.

@Lipt0nas could also give his input :)

Reply:
I initially planned to drop the connection and `governor` actually has this feature but it has a flaw that it won't let in any request past the limit. Meaning that if we allow for example 50 requests per sec and somebody spams with 51 per second, no request is going to be handled whatsoever (I imagined that 50 of then still would be handled). Do you think that's OK?

I agree about the limit

Reply:
Ok, so I was wrong. As it turns out it will pass this 50 of 51 requests

Reply:
Fixed. Now it will respond with 429 when the limit is reached